parts/linux/cloud-init/artifacts/reconcile-private-hosts.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+set -o nounset
+set -o pipefail
+SLEEP_SECONDS=15
+clusterFQDN={{GetParameter "kubernetesEndpoint"}}
+if [[ $clusterFQDN != *.privatelink.* ]]; then
+  echo "skip reconcile hosts for $clusterFQDN since it's not AKS private cluster"
+  exit 0
+fi
+echo "clusterFQDN: $clusterFQDN"
+function get-apiserver-ip-from-tags() {
+  tags=$(curl -sSL -H "Metadata: true" "http://169.254.169.254/metadata/instance/compute/tags?api-version=2019-03-11&format=text")
+  if [ "$?" == "0" ]; then
+    IFS=";" read -ra tagList <<< "$tags"
+    for i in "${tagList[@]}"; do
+      tagKey=$(cut -d":" -f1 <<<$i)
+      tagValue=$(cut -d":" -f2 <<<$i)
+      if [ "$tagKey" == "aksAPIServerIPAddress" ]; then
+        echo -n "$tagValue"
+        return
+      fi
+    done
+  fi
+  echo -n ""
+}
+
+while true; do
+  clusterIP=$(get-apiserver-ip-from-tags)
+  if [ -z $clusterIP ]; then
+    sleep "${SLEEP_SECONDS}"
+    continue
+  fi
+  if grep "$clusterIP $clusterFQDN" /etc/hosts; then
+    echo "$clusterFQDN has already been set to $clusterIP"
+  else
+    sudo sed -i "/$clusterFQDN/d" /etc/hosts
+    sudo sed -i "\$a$clusterIP $clusterFQDN" /etc/hosts
```

Comment:
what does $a mean here? Why not just append a new line
echo "$clusterIP $clusterFQDN" >> /etc/hosts?

Reply:
$a just do the append. I'll change to echo for less misleading.

Reply:
Got you. There are other (better) ways, sudo tee or sudo bash -c.

https://www.cyberciti.biz/faq/sudo-append-data-text-to-file-on-linux-unix-macos/

Reply:
thanks, used tee style.